mm2src/coins/lp_coins.rs

```diff
@@ -2296,9 +2296,17 @@ pub enum ValidatorsInfoDetails {
 }
 
 #[derive(Serialize, Deserialize)]
-pub struct SignatureRequest {
-    coin: String,
-    message: String,
+#[serde(tag = "derivation_method", rename_all = "snake_case")]
+pub enum SignatureRequest {
+    Iguana {
+        coin: String,
+        message: String,
+    },
+    HdWallet {
+        coin: String,
+        message: String,
+        derivation_path: RpcDerivationPath,
```

Comment:
there is part of the derivation path which should be static per coin (and defined in the coins file). 
we might wanna get the derivation path this way: https://github.com/KomodoPlatform/komodo-defi-framework/blob/d323812a5100b3a96f152bfcbc7ac38e86b828bc/mm2src/coins/rpc_command/account_balance.rs#L19-L20

Reply:
I might disagree with this. KDF already returns full derivation path for every addresses which `GUI` can just pass to this param. Moreover, if I go  with your suggestion, I will also need to make `sign_message` into asynchronous function(very small thing tho) and make the fn do more work.

If I see a better reason not to require the full `derivation_path` I will rework it ofc.

https://github.com/KomodoPlatform/komodo-defi-framework/issues/2406#issuecomment-2834679879

Reply:
if we wanna keep it like this, we at least wanna perform a check ([e.g.](https://github.com/KomodoPlatform/komodo-defi-framework/blob/d323812a5100b3a96f152bfcbc7ac38e86b828bc/mm2src/coins/tendermint/tendermint_coin.rs#L1200)) to make sure we are not signing from a private key that isn't standard to this coin. people make mistakes.

Reply:
done